Add option to migrate to PHP based grid configuration

mamazu opened the pull request:
| Q               | A
|-----------------|-----
| Branch?         | 2.3
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | yes
| Related tickets | closes #18136 
| License         | MIT

Adding alternative grid configurations in php so that in Sylius 3.0 we can switch to them and remove the yaml configs.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

* **New Features**
  * Added configurable grid definitions using PHP, allowing flexible configuration alongside YAML options.
  * Added ability to override individual grid configurations with per-grid or global settings.

* **Chores**
  * Updated Grid Bundle dependency from ^1.13 to ^1.14.
  * Added architecture rules to enforce consistent grid implementations.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mamazu committed:
Applying the shop grids

Paritally copying over some of the grids from #14449

mamazu committed:
Fixing the errors?

mamazu committed:
Adding dirty fix for test

mamazu committed:
Fixing errors and adding label back

mamazu committed:
Apply suggestions from code review

Co-authored-by: Dmitri Perunov <diimpp@gmail.com>

mamazu committed:
Using the right actions

loic425 committed:
Configurable grid provider

mamazu committed:
Removing custom template

mamazu committed:
Adding upgrade guide

mamazu committed:
Fixing the config defintion and the upgrade docs

mamazu committed:
Update src/Sylius/Bundle/CoreBundle/Grid/Provider/ConfigurableProvider.php

Co-authored-by: Dmitri Perunov <diimpp@gmail.com>

mamazu committed:
Refactoring the code

mamazu committed:
Constrain it to only those two values

mamazu committed:
Making the default value also an enum node

mamazu committed:
Moving the documentation

mamazu committed:
Merge remote-tracking branch 'upstream/2.1' into feature/configurable-grid-provider

mamazu committed:
Update src/Sylius/Bundle/CoreBundle/Tests/Grid/Provider/ConfigurableProviderTest.php

Co-authored-by: Loïc Frémont <lc.fremont@gmail.com>

mamazu committed:
Merge branch '2.1' into feature/configurable-grid-provider

mamazu committed:
Merge branch 'php-grid-configuration' into feature/configurable-grid-provider

mamazu committed:
Update src/Sylius/Bundle/CoreBundle/Resources/config/services/providers.xml

Co-authored-by: Dmitri Perunov <diimpp@gmail.com>

mamazu committed:
Using an override

mamazu committed:
Fixing the service config

mamazu committed:
Fixing stuff

mamazu committed:
Configuring symfony DI

mamazu committed:
Fixing the grids based on the new template paths

mamazu committed:
Removing custom datetimeformat

mamazu committed:
Adding a test to check if the service and yaml configuration are the same

mamazu committed:
Removing yaml from the name of the configuration key

mamazu committed:
Configurable grid provider (#17675)

| Q               | A
|-----------------|-----
| Branch?         | 2.1
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | mentioned in
https://github.com/Sylius/Sylius/pull/15419, replaces: #15491
| License         | MIT

The same pull request from @loic425 but rebased to 2.0 and some fixes in
the `Configuration` class so that it works as described in the PR (and
upgrade docs).


<!-- This is an auto-generated comment: release notes by coderabbit.ai
-->
## Summary by CodeRabbit

- **New Features**
- Introduced a configurable grid provider system that dynamically
selects grid services based on configuration.
- Added new configuration options for setting default grid types and
defining grid services per grid code.

- **Documentation**
- Updated upgrade notes to include guidance on switching between PHP and
YAML formats for grid configurations, with example snippets.

- **Tests**
- Added comprehensive unit tests for the `ConfigurableProvider` class to
verify correct provider selection and error handling.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mamazu committed:
Apply suggestions from code review

Co-authored-by: Loïc Frémont <lc.fremont@gmail.com>

mamazu committed:
Migrating the ShopBundle/ProductGrid

mamazu committed:
Migrating the ShopBundle/Account/OrderGrid

mamazu committed:
Using the expression language

mamazu committed:
Merge remote-tracking branch 'upstream/2.1' into php_grids_for_shop_bundle

mamazu committed:
Merge remote-tracking branch 'upstream/2.1' into php-grid-configuration

mamazu committed:
Merge remote-tracking branch 'upstream/php-grid-configuration' into php_grids_for_shop_bundle

mamazu committed:
Apply suggestions from code review

Co-authored-by: Loïc Frémont <lc.fremont@gmail.com>

loic425 committed:
Adding grid configuration in php for the ShopBundle (#15419)

Paritally copying over some of the grids from #14449

| Q | A |

|-----------------|--------------------------------------------------------------|
| Branch? | 1.13 <!-- see the comment below --> |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no <!-- don't forget to update the UPGRADE-*.md file
--> |
| Related tickets | partially #14449                      |
| License | MIT |

To be able to get this merged quicker this MR contains the shop grid
bundle changes only for the shop.

mamazu committed:
Reverting grid to standard

mamazu committed:
Grid configuration test (#18188)

| Q               | A
|-----------------|-----
| Branch?         | 2.1 <!-- see the comment below -->
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no-
| Related tickets | based on 15419
| License         | MIT

## Contents

Adding a test that checks if the php service grids are the same as the
ones in the yaml configuration.

mamazu committed:
Admin user grid

mamazu committed:
Grid service config

mamazu committed:
Fixing up some templates

mamazu committed:
Improving the grid tests

mamazu committed:
Merge branch '2.1' into php-grid-configuration

mamazu committed:
Moving it to the correct place

mamazu committed:
Fixing some inconsistencies

mamazu committed:
Testing all new php grids if migrated (#18516)

# What I did

* Replace the one test with a loop with a dataprovider (then it doesn't
crash on the first error but checks all grids)
* Skipping the tests of unmigrated grids (nice to track the progress)
* Using the existing grids in Sylius as point of reference (not using
the config for migrating stuff, but all sylius defined grids)

## Problems
There are some grids that work differently with php than in YAML. How do
we want to deal with that?

cc @loic425

mamazu committed:
Codestyle

mamazu committed:
Configure Grid Catalog Promotions

loic425 committed:
Prefill action labels

mamazu committed:
Apply suggestion from @loic425

Co-authored-by: Loïc Frémont <lc.fremont@gmail.com>

mamazu committed:
Configuring `AdminUserGrid` with PHP (#18489)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | #18136
| License         | MIT

Replacing the admin grid with php configuration

mamazu committed:
Merge branch 'php-grid-configuration' into grid-catalog-promotion

mamazu committed:
Fixing tests

mamazu committed:
Configure `CatalogPromotionGrid` with PHP (#18519)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Moved the grids in the same namespace

mamazu committed:
PHP Grid configuration for `ChannelGrid`

mamazu committed:
Apply suggestion from @loic425

Co-authored-by: Loïc Frémont <lc.fremont@gmail.com>

mamazu committed:
Php config for `ChannelGrid` (#18521)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Currency grid

mamazu committed:
Codestyle

mamazu committed:
Remove interface

mamazu committed:
Migrating customer grid

mamazu committed:
fixups

mamazu committed:
Migrating `CustomerGrid` (#18522)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Migrating CustomerGroupGrid

mamazu committed:
Cusomter Order and Order Grid

mamazu committed:
Cusomter Order and Order Grid

mamazu committed:
Migrating php inventory grid

mamazu committed:
Migrating LocaleGrid to php config

mamazu committed:
Migrating PaymentGrid

mamazu committed:
Migrate PaymentMethod to php grid

mamazu committed:
Migrating ProductAssociationGrid to php

mamazu committed:
Migrate ProductAttribute Grid

mamazu committed:
Product grids

crydotsnake committed:
TASK: [AdminBundle] add PHP configuration for country grid

crydotsnake committed:
TASK: apply review feedback

mamazu committed:
Using CreateAction instead of manually creating the action

mamazu committed:
Updating configuration to match the old grid

mamazu committed:
Using string filter instead of generic Filter

mamazu committed:
Migrating ProductOptions

mamazu committed:
Migrate ProductAttribute Grid (#18530)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Code Review suggestions

mamazu committed:
Migrating `ProductGrid` and `ProductFromTaxon` (#18531)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Migrating ProductReview grid

mamazu committed:
Migrate ProductVariantGrid

mamazu committed:
TASK: [AdminBundle] add PHP configuration for country grid (#18532)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | part of #18136 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.14 or 2.0 branch
 - Features and deprecations must be submitted against the 2.1 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

mamazu committed:
Migrate ProductVariantWithCatalogPromotionsGrid

mamazu committed:
Migrating PromotionGrid

mamazu committed:
Migrating the ShippingGrids

mamazu committed:
Migrating Tax*Grid

crydotsnake committed:
TASK: [AdminBundle] migrate ZoneGrid too PHP

mamazu committed:
Merge branch 'php-grid-configuration' into product-association-grid-php

mamazu committed:
Migrating ProductAssociationGrid to php (#18529)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Merge branch 'php-grid-configuration' into customer-group-grid

mamazu committed:
Update src/Sylius/Bundle/AdminBundle/Grid/PaymentMethodGrid.php

Co-authored-by: Loïc Frémont <lc.fremont@gmail.com>

mamazu committed:
Update src/Sylius/Bundle/AdminBundle/Grid/PaymentMethodGrid.php

Co-authored-by: Loïc Frémont <lc.fremont@gmail.com>

mamazu committed:
Merge branch 'php-grid-configuration' into payment-method-grid-php

mamazu committed:
Migrate PaymentMethod to php grid (#18528)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Merge branch 'php-grid-configuration' into customer-group-grid

mamazu committed:
Merge branch 'php-grid-configuration' into php-inventory-grid

mamazu committed:
Php config for `InventoryGrid` (#18525)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Update src/Sylius/Bundle/AdminBundle/Grid/LocaleGrid.php

Co-authored-by: Loïc Frémont <lc.fremont@gmail.com>

mamazu committed:
Merge branch 'php-grid-configuration' into php-locale-grid

mamazu committed:
Migrating LocaleGrid to php config (#18526)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

crydotsnake committed:
TASK: add StringFilter for name and code and apply phpcs fixes

mamazu committed:
TASK: [AdminBundle] migrate ZoneGrid to PHP (#18543)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | part of #18136 
| License         | MIT

mamazu committed:
Merge branch 'php-grid-configuration' into customer-group-grid

mamazu committed:
Customer group grid (#18523)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Merge remote-tracking branch 'upstream/2.2' into php-grid-configuration

mamazu committed:
Removing resource aware grid

mamazu committed:
Apply suggestions from code review

Co-authored-by: Loïc Frémont <lc.fremont@gmail.com>

mamazu committed:
Apply suggestions from code review

Co-authored-by: Loïc Frémont <lc.fremont@gmail.com>

mamazu committed:
Fixing tests

mamazu committed:
Migrating PaymentGrid (#18527)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Customer Order and Order Grid (#18524)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Fixing tests

mamazu committed:
Fixing tests with kernel shutdown (#18558)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Merge branch 'php-grid-configuration' into tax-grids-php-config

mamazu committed:
Merge branch 'php-grid-configuration' into shipping-grids-php

mamazu committed:
Merge branch 'php-grid-configuration' into promotion-grid-php

mamazu committed:
Merge branch 'php-grid-configuration' into product_variant_grid-php

mamazu committed:
Using the proper filters

mamazu committed:
Merge branch 'php-grid-configuration' into product-options-php-grid

mamazu committed:
Migrating PromotionGrid (#18538)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Merge branch 'php-grid-configuration' into tax-grids-php-config

mamazu committed:
Migrating Tax*Grid (#18540)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Migrating ProductReview grid (#18535)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Merge branch 'php-grid-configuration' into product-options-php-grid

mamazu committed:
Migrating ProductOptions (#18533)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Migrate ExchangeRateGrid

mamazu committed:
Merge branch 'php-grid-configuration' into shipping-grids-php

mamazu committed:
Migrating the ShippingGrids (#18539)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Skipping deprecated sylius_admin_address_log_entry grid

mamazu committed:
Migrating ChannelPricingGrids to php

mamazu committed:
Migrating PaymentRequestGrid to php config

mamazu committed:
Migrating ChannelPricingGrids to php (#18560)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Migrating PaymentRequestGrid to php config (#18561)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Migrate ExchangeRateGrid (#18559)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Migrate ProductVariantWithCatalogPromotionsGrid (#18537)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

mamazu committed:
Merge branch 'php-grid-configuration' into product_variant_grid-php

mamazu committed:
Migrating the ProductFromTaxonGrid

mamazu committed:
Migrate ProductVariantGrid (#18536)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configruation
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | rel #18136
| License         | MIT

loic425 committed:
Improve admin grids

loic425 committed:
Improve admin grids (#18575)

| Q               | A

|-----------------|--------------------------------------------------------------
| Branch? | php-grid-configuration <!-- see the comment below -->
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations? | no <!-- don't forget to update the UPGRADE-*.md file
-->
| Related tickets | partially #18136
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.14 or 2.0 branch
 - Features and deprecations must be submitted against the 2.1 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

mamazu commented on UPGRADE-2.1.md:
This would probably have to be moved now to 2.2.

loic425 committed:
Use invokable grids

loic425 committed:
Use invokable grids (#18578)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |partially #18136 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.14 or 2.0 branch
 - Features and deprecations must be submitted against the 2.1 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

```php
#[AsDecorator('sylius_admin.grid.product')]
#[AsGrid(name: 'sylius_admin_product')]
final class AdminProductGrid extends AbstractGrid implements ProductGridInterface
{
    public function __construct(private readonly ProductGridInterface $productGrid)
    {
    }

    public function __invoke(GridBuilderInterface $gridBuilder): void
    {
        ($this->productGrid)($gridBuilder);

        $gridBuilder
            ->orderBy('name', 'desc')
            ->removeField('code')
            ->setLimits([5, 10])
        ;
    }
}
```

loic425 committed:
Improve shop grids

loic425 committed:
Improve shop grids (#18585)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |partially #18136 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.14 or 2.0 branch
 - Features and deprecations must be submitted against the 2.1 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

loic425 committed:
[PhpGridConfiguration] Remove coupling interfaces to grid ones

loic425 committed:
[php-grid-configuration] Remove coupling interfaces to grid ones (#18598)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |  partially #18136
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.14 or 2.0 branch
 - Features and deprecations must be submitted against the 2.1 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

The Grids were still relying on an interface tightly coupled to the Grid
system, creating unnecessary dependencies between the Sylius interfaces
and the Grid one.
When decorating a grid, only the __invoke() method needs to be called,
making the extra interface redundant.

### ✅ What’s changed

Removed a Grid-specific interface that introduced unnecessary coupling.

### 🎯 Motivation

Improves separation of concerns by making grids more independent from
specific Grid abstractions.

Reduces rigid coupling and aligns the implementation with cleaner, more
flexible architecture principles.

### 📎 Related

Partially addresses issue #18136.

mamazu committed:
Merge remote-tracking branch 'upstream/2.2' into php-grid-configuration

mamazu committed:
Reverting composer.json changes in the UI bundle

mamazu committed:
Moving the changelog to the correct version

mamazu committed:
Upgrading packages

mamazu committed:
Merge branch '2.3' into php-grid-configuration

Moving grid service configuration to php

mamazu committed:
Codestyle

mamazu committed:
Fixing ProductTaxon

mamazu commented on src/Sylius/Bundle/AdminBundle/Grid/PaymentMethodGrid.php:
It's the same as in the yaml config

mamazu committed:
Implementing code review suggestions

mamazu committed:
Enforcing grids to be final

mamazu committed:
Moving the changelog to the correct version

mamazu commented on UPGRADE-2.3.md:
Maybe we should also explain a little bit of the reasoning about it, and point to some documentation.

Like:
- In the future Sylius will drop Yaml config for grids
- Upgrade path is using a grid migrator or AI to replace your projects Yaml with php
- For more info on how to extend the php grids: go to the grid docs

mamazu committed:
Merge remote-tracking branch 'upstream/2.3' into php-grid-configuration

mamazu committed:
Finishing the PR with all grids

GSadee commented on UPGRADE-2.3.md:
@mamazu, fully agree, let's expand the note along those lines to include:

- The reasoning & benefits: why we're moving grid config from YAML to PHP
- The 3.0 direction
- The upgrade path with an example of customization in the end app
- Tooling: I guess you can point to https://github.com/mamazu/grid-config-converter to automate the conversion
- Docs: a link to Grid docs and maybe even a dedicated page for migration

GSadee commented on src/Sylius/Bundle/AdminBundle/Grid/CountryGrid.php:
There is some inconsistency, because some of classes use `AsGrid` attribute and some `getName` method to name the grid

GSadee commented on UPGRADE-2.3.md:
Shouldn't it be `use_legacy_config` with the default null value?

What is more, could you clarify here the precedence between the global `use_legacy_config` and the per-grid one, because currently a non-null global value overrides every per-grid setting, so "switch all to PHP globally but keep one grid on YAML" isn't possible, which is worth documenting (or reconsidering)?

GSadee reviewed (commented):
Could we add at least one CI build that runs the Behat suite with `sylius_core.grid.use_legacy_config: false`? Right now the PHP grids are only covered structurally by OverridenGridsTest, nothing checks them behaviorally (admin/shop rendering, filtering, sorting), since every Behat run defaults to the YAML path.

mamazu committed:
Replacing getName with attributes

mamazu committed:
Reworking the upgrade guide

mamazu commented on UPGRADE-2.3.md:
I've updated the documentation to include all of it. Can you have a look again?

mamazu commented on UPGRADE-2.3.md:
The updated docs should make that clearer now. I think.

But as to your Behat suggestion, that does sound like a good idea. Although I don't expect any tests to fail, I'd like to but I don't know what the best approach for this would be.

loic425 committed:
feat(php-grids): Remove abstract grid usage

mamazu committed:
Making grid assertions explicit

mamazu committed:
Ordering the keys correctly

GSadee committed:
feat(php-grids): Remove abstract grid usage (#19070)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations? | no <!-- don't forget to update the UPGRADE-*.md file
-->
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 2.2 branch
 - Features and deprecations must be submitted against the 2.3 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

loic425 committed:
Making grid assertions explicit (#19077)

| Q               | A
|-----------------|-----
| Branch?         | 2.2
| Bug fix?        | no
| New feature?    | -
| BC breaks?      | -
| Deprecations?   | -
| Related tickets | -
| License         | MIT

Making the assertions explicit and fixing the issues found with this.

<!-- This is an auto-generated comment: release notes by coderabbit.ai
-->

## Summary by CodeRabbit

## Release Notes

* **Refactor**
* Reorganized column and filter ordering in admin grids for improved
layout consistency across Product, Promotion, Shipping Method, and
Catalog Promotion interfaces.
* Applied default filter values to archival filters to improve initial
filtering behavior.
  * Enhanced grid testing infrastructure for better comparison accuracy.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

loic425 committed:
feat(php-grids): Tests with Legacy grid config

# Conflicts:
#	.github/workflows/ci_e2e-mariadb.yaml
#	.github/workflows/matrix.json

mamazu committed:
Merge branch '2.3' into php-grid-configuration

loic425 committed:
feat(php-grids): Use interfaces for grid names

loic425 committed:
feat(php-grids): Use interfaces for grid names (#19097)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations? | no <!-- don't forget to update the UPGRADE-*.md file
-->
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 2.2 branch
 - Features and deprecations must be submitted against the 2.3 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

loic425 commented on UPGRADE-2.3.md:
I think this part is ok now. And the CI has been improved in https://github.com/Sylius/Sylius/pull/19091

loic425 commented on UPGRADE-2.3.md:
We also need to update the documentation with Grid mutators now they are ready.

NoResponseMate committed:
feat(php-grids): Tests with Legacy grid config (#19091)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations? | no <!-- don't forget to update the UPGRADE-*.md file
-->
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 2.2 branch
 - Features and deprecations must be submitted against the 2.3 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

mamazu commented on UPGRADE-2.3.md:
Yeah and they should also be covered by the grid config converter.

loic425 committed:
feat(php-grids): Use withFields and withFilters

loic425 committed:
feat(php-grids): Use withFields and withFilters (#19099)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations? | no <!-- don't forget to update the UPGRADE-*.md file
-->
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 2.2 branch
 - Features and deprecations must be submitted against the 2.3 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

loic425 committed:
feat(php-grids): Use new with actions' helpers

loic425 committed:
feat(php-grids): use resourceClass option in AsGrid attribute

loic425 committed:
feat(php-grids): Use new with actions' helpers (#19100)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations? | no <!-- don't forget to update the UPGRADE-*.md file
-->
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 2.2 branch
 - Features and deprecations must be submitted against the 2.3 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

loic425 committed:
feat(php-grids): use resourceClass option in AsGrid attribute (#19102)

| Q               | A
|-----------------|-----
| Branch?         | php-grid-configuration
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations? | no <!-- don't forget to update the UPGRADE-*.md file
-->
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 2.2 branch
 - Features and deprecations must be submitted against the 2.3 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

GSadee commented on src/Sylius/Bundle/ShopBundle/Resources/config/services/grid.php:
It should be imported above
```suggestion
    $services->set('sylius_shop.grid.account.order', OrderGrid::class)
```

GSadee commented on composer.json:
Why not 1.16 as well? 🤔

GSadee commented on UPGRADE-2.3.md:
```suggestion
* [Grid Config Converter](https://github.com/mamazu/grid-config-converter)
```
